src/immutable_data.rs

```diff
@@ -38,7 +38,7 @@ impl UnpubImmutableData {
         // TODO: Use low-level arrays or slices instead of Vec.
         let mut bytes = Vec::with_capacity(XOR_NAME_LEN + PK_SIZE);
         bytes.extend_from_slice(&tiny_keccak::sha3_256(&value));
-        bytes.extend_from_slice(&owners.to_bytes());
+        bytes.extend_from_slice(&owners.to_bytes()[..48]);
```

Comment:
Where does `48` come from? Can we use a descriptive constant here instead of hard-coding a number?

Reply:
I think rather than writing a new `to_bytes()` method we should just use the existing ability to serialise the `PublicKey`.  Deriving the address could be something like:
```rust
let hash_of_value = tiny_keccak::sha3_256(&value);
let serialised_contents = utils::serialise(&(hash_of_value, &owners));
let address = Address::Unpub(XorName(tiny_keccak::sha3_256(&serialised_contents)));
```